go/vt/vtctl/schematools/schematools.go

```diff
@@ -43,3 +47,66 @@ func GetSchema(ctx context.Context, ts *topo.Server, tmc tmclient.TabletManagerC
 
 	return sd, nil
 }
+
+// ParseSchemaMigrationStrategy parses the given strategy into the underlying enum type.
+func ParseSchemaMigrationStrategy(name string) (vtctldatapb.SchemaMigration_Strategy, error) {
+	if name == "" {
+		// backward compatiblity and to handle unspecified values
+		return vtctldatapb.SchemaMigration_DIRECT, nil
+
+	}
+
+	upperName := strings.ToUpper(name)
+	switch upperName {
+	case "GH-OST", "PT-OSC":
+		// more backward compatibility since the protobuf message names don't
+		// have the dash.
+		upperName = strings.ReplaceAll(upperName, "-", "")
+	default:
+	}
+
+	strategy, ok := vtctldatapb.SchemaMigration_Strategy_value[upperName]
+	if !ok {
+		return 0, fmt.Errorf("unknown schema migration strategy: '%v'", name)
+	}
+
+	if upperName != name {
+		// TODO (andrew): Remove special handling for lower/uppercase and
+		// gh-ost=>ghost/pt-osc=>ptosc support. (file issue for this).
+		log.Warningf("detected legacy strategy name syntax; parsed %q as %q. this will break in the next version.", upperName, name)
```

Comment:
i actually think we maybe don't want to warn at all, and instead accept both (since to truly deprecate this we'd also have to change the strings that `vttablet` uses to write to the table)

Reply:
Just offering my take, people should use `gh-ost` and `pt-osc` and we don't need to cover for spelling mistakes or upper/lower.

Reply:
Yep, I'm removing these warnings now. My question now though is, should we update `SchemaMigrationStrategyName` to re-dash the GHOST and PTOSC enums? In other words, should this pass?

```go
assert.Equal(t, "gh-ost", SchemaMigrationStrategyName(vtctldatapb.SchemaMigrationStrategy_GHOST)
assert.Equal(t, "pt-osc", SchemaMigrationStrategyName(vtctldatapb.SchemaMigrationStrategy_PTOSC)
```

Reply:
Yes, I think this should pass. `gh-ost` is the official name, `ghost` never was. And we choose the same notation for `pt-osc` (FYI: we will deprecate support for `pt-osc` sometime soon).

Reply:
https://github.com/vitessio/vitess/pull/13738/commits/8e58aecf3dd8d94b9c85b8fa18c00d3542bb4d48